app/Services/ReservationReassignmentService.php

```
@@ -167,56 +138,155 @@ public function reassignOnNewCopy(int $libroId, int $newCopiaId): void
+                $hasOverlap = false;
+                foreach ($targetCommitments as $commitment) {
+                    if ((int) $commitment['id'] === (int) $candidate['id']) {
+                        continue;
+                    }
+
+                    if ((int) $commitment['utente_id'] === (int) $candidate['utente_id']) {
+                        $identityConflict = true;
+                        break;
+                    }
+
+                    $startsBeforeCandidateEnds = (string) $commitment['data_prestito']
```

PHPMD (2.15.0)

[warning] Avoid excessively long variable names like $startsBeforeCandidateEnds. Keep variable name length under 20. (undefined)

(LongVariable)